sormas-backend/src/main/java/de/symeda/sormas/backend/action/ActionService.java

```diff
@@ -0,0 +1,220 @@
+import de.symeda.sormas.backend.event.EventService;
+import de.symeda.sormas.backend.user.User;
+
+@Stateless
+@LocalBean
+public class ActionService extends AbstractAdoService<Action> {
+
+	@EJB
+	private EventService eventService;
+
+	public ActionService() {
+		super(Action.class);
+	}
+
+	public List<Action> getAllActiveActionsAfter(Date date, User user) {
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<Action> cq = cb.createQuery(getElementClass());
+		Root<Action> from = cq.from(getElementClass());
+		Predicate filter = buildActiveActionsFilter(cb, from);
+		if (user != null) {
+			Predicate userFilter = createUserFilter(cb, cq, from);
+			filter = AbstractAdoService.and(cb, filter, userFilter);
+		}
+		if (date != null) {
+			Predicate dateFilter = createChangeDateFilter(cb, from, date);
+			filter = AbstractAdoService.and(cb, filter, dateFilter);
+		}
+		cq.where(filter);
+		cq.orderBy(cb.desc(from.get(Action.CHANGE_DATE)));
+		cq.distinct(true);
+		return em.createQuery(cq).getResultList();
+	}
+
+	public List<String> getAllActiveUuids(User user) {
+
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<String> cq = cb.createQuery(String.class);
+		Root<Action> from = cq.from(getElementClass());
+
+		Predicate filter = buildActiveActionsFilter(cb, from);
+
+		if (user != null) {
+			Predicate userFilter = createUserFilter(cb, cq, from);
+			filter = AbstractAdoService.and(cb, filter, userFilter);
+		}
+
+		cq.where(filter);
+		cq.select(from.get(Action.UUID));
+
+		return em.createQuery(cq).getResultList();
+	}
+
+	public List<Action> getAllByEvent(Event event) {
+
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<Action> cq = cb.createQuery(getElementClass());
+		Root<Action> from = cq.from(getElementClass());
+
+		Predicate filter = cb.equal(from.get(EventParticipant.EVENT), event);
+		cq.where(filter);
+		cq.orderBy(cb.desc(from.get(EventParticipant.CREATION_DATE)));
+
+		return em.createQuery(cq).getResultList();
+	}
+
+	/**
+	 * @see /sormas-backend/doc/UserDataAccess.md
+	 */
+	@SuppressWarnings("rawtypes")
+	@Override
+	public Predicate createUserFilter(CriteriaBuilder cb, CriteriaQuery cq, From<Action, Action> actionPath) {
+
+		// National users can access all actions in the system
+		User currentUser = getCurrentUser();
+		if (currentUser.hasAnyUserRole(UserRole.NATIONAL_USER, UserRole.NATIONAL_CLINICIAN, UserRole.NATIONAL_OBSERVER, UserRole.REST_USER)) {
+			return null;
+		}
+
+		// whoever created the action is allowed to access it
+		Predicate filter = cb.equal(actionPath.join(Action.CREATOR_USER, JoinType.LEFT), currentUser);
+
+		Predicate eventFilter = eventService.createUserFilter(cb, cq, actionPath.join(Action.EVENT, JoinType.LEFT));
+		if (eventFilter != null) {
+			filter = cb.or(filter, eventFilter);
+		}
+
+		return filter;
+	}
+
+	private Predicate buildActiveActionsFilter(CriteriaBuilder cb, Root<Action> from) {
```

Comment:
Since there's no global list of actions, this shouldn't be necessary

Reply:
It's used by the rest API.

Reply:
Maybe the api is not needed, what do you think ?

Reply:
Right now it's not necessary because there's no actions in the mobile app; it might be necessary in the future though, so I'd leave it there. I don't see where it's needed in the ReST API though - where did you see a reference?

Reply:
`sormas-rest/src/main/java/de/symeda/sormas/rest/ActionResource.java` called `ActionFacade.getAllActive*`

https://github.com/hzi-braunschweig/SORMAS-Project/compare/bb009ad25ae0aca4b3f6d4428afea021406bd5ad..abdaf021c3a55d01c34488bc3fe5d02dfd18c5cc#diff-69bad65e268bb21f35a5fc9120cd0635